core/deployment/src/main/java/io/quarkus/artemis/core/deployment/ArtemisCoreProcessor.java

```diff
@@ -2,112 +2,126 @@
 
+import org.apache.activemq.artemis.api.core.ActiveMQBuffer;
 import org.apache.activemq.artemis.api.core.client.ServerLocator;
-import org.apache.activemq.artemis.api.core.client.loadbalance.ConnectionLoadBalancingPolicy;
-import org.apache.activemq.artemis.api.core.client.loadbalance.FirstElementConnectionLoadBalancingPolicy;
-import org.apache.activemq.artemis.api.core.client.loadbalance.RandomConnectionLoadBalancingPolicy;
-import org.apache.activemq.artemis.api.core.client.loadbalance.RandomStickyConnectionLoadBalancingPolicy;
-import org.apache.activemq.artemis.api.core.client.loadbalance.RoundRobinConnectionLoadBalancingPolicy;
+import org.apache.activemq.artemis.api.core.client.loadbalance.*;
 import org.apache.activemq.artemis.core.remoting.impl.netty.NettyConnectorFactory;
 import org.apache.activemq.artemis.spi.core.remoting.ConnectorFactory;
+import org.apache.activemq.artemis.utils.RandomUtil;
 import org.jboss.jandex.ClassInfo;
 import org.jboss.jandex.DotName;
 import org.jboss.logging.Logger;
 
 import io.quarkus.arc.deployment.SyntheticBeanBuildItem;
+import io.quarkus.artemis.core.common.ArtemisUtil;
+import io.quarkus.artemis.core.runtime.ArtemisBuildTimeConfigs;
 import io.quarkus.artemis.core.runtime.ArtemisCoreRecorder;
-import io.quarkus.deployment.annotations.BuildProducer;
-import io.quarkus.deployment.annotations.BuildStep;
-import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.artemis.core.runtime.ArtemisRuntimeConfigs;
+import io.quarkus.deployment.annotations.*;
 import io.quarkus.deployment.annotations.Record;
 import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.NativeImageConfigBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
-import io.quarkus.smallrye.health.deployment.spi.HealthBuildItem;
 
 public class ArtemisCoreProcessor {
-
     private static final Logger LOGGER = Logger.getLogger(ArtemisCoreProcessor.class);
 
-    static final Class[] BUILTIN_CONNECTOR_FACTORIES = {
+    static final Class<?>[] BUILTIN_CONNECTOR_FACTORIES = {
             NettyConnectorFactory.class
     };
 
-    static final Class[] BUILTIN_LOADBALANCING_POLICIES = {
+    static final Class<?>[] BUILTIN_LOADBALANCING_POLICIES = {
             FirstElementConnectionLoadBalancingPolicy.class,
             RandomConnectionLoadBalancingPolicy.class,
             RandomStickyConnectionLoadBalancingPolicy.class,
             RoundRobinConnectionLoadBalancingPolicy.class
     };
+    protected static final String FEATURE = "artemis-core";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
 
     @BuildStep
     NativeImageConfigBuildItem config() {
         return NativeImageConfigBuildItem.builder()
-                .addRuntimeInitializedClass("org.apache.activemq.artemis.api.core.ActiveMQBuffers")
```

Comment:
I thought there was a reason to use strings, does this work?

Reply:
We are still calling it with a `String`, since we call `...class.getCanonicalName()`

Reply:
Can this be resolved?

Reply:
Yes, I think pipeline will break if it doesn't work.